### PR TITLE
VIC submit with polling

### PR DESCRIPTION
**Pull request by jbalboni**
Sets up submit call and polls for completion.

I added the ability to specify a submit function in the form config that returns a promise and completely skip the normal submit code. I don't know if this way of polling will be something other forms will need to do in a consistent enough way to build that into our common schemaform code.

### CODE REVIEW

**Review by cehsu — APPROVED** [review]
LGTM. I had a few minor questions.

**Review by rianfowler — APPROVED** [review]

File: src/js/vic-v2/helpers.js [hunk 1]
```diff
@@ -29,3 +32,81 @@ export function prefillTransformer(pages, formData, metadata) {
...
+      if (res instanceof Error) {
+        Raven.captureException(res);
+        Raven.captureMessage('vets_vic_poll_client_error');
+
+        // keep polling because we know they submitted earlier
+        // and this is likely a network error
+        return Promise.resolve();
+      }
+
+      // if we get here, it's likely that we hit a server error
```
[the remark this answers is not in the document]
Comment (jbalboni) [reply src/js/vic-v2/helpers.js]: It should be captured in the backend Sentry logging. We sometimes capture those on the front end, but I think we should move away from it.

File: src/js/vic-v2/helpers.js [hunk 2]
```diff
@@ -29,3 +32,81 @@ export function prefillTransformer(pages, formData, metadata) {
...
+      return Promise.reject(res);
+    });
+}
+
+const POLLING_INTERVAL = 1000;
```
[the remark this answers is not in the document]
Comment (jbalboni) [reply src/js/vic-v2/helpers.js]: Just a rough guess. We can adjust it if it seems wrong during e2e testing.

File: src/js/vic-v2/helpers.js [hunk 3]
```diff
@@ -29,3 +32,81 @@ export function prefillTransformer(pages, formData, metadata) {
...
+      method: 'POST',
+      headers,
+      body
+    }).then((res) => {
+      if (res.ok) {
+        return res.json();
+      }
+
+      return Promise.reject(res);
+    }).then(resp => {
```
[the remark this answers is not in the document]
Comment (jbalboni) [reply src/js/vic-v2/helpers.js]: I think `res` or `resp` makes sense everywhere and we should try to use that. But maybe not something it's worth spending time on making consistent across teams.

**Pull request merged**